### PR TITLE
Add exception to bounds catch for l.crs.simple

### LINKED ISSUE #1007: Some handlers not working ... after fix #517
Good morning,

for some reason drag, rotate, freerotate and other handlers won't do his job in some circumstances.

After a lot of time debugging the code, to find the problem I have noticed that changes in #517 has broke the usual behavior of this handlers.

I have noticed that the problem is in the function:

`  _cornerExceedsMapLats: function(zoom, corner, map) {...}` on file DistortabeImageOverlay.js
 
 and the problem becomes in the next check:

```
    if (zoom === 0) {
      exceedsTop = map.project(corner).y < 2;
      exceedsBottom = map.project(corner).y >= 255;
    } else {
      exceedsTop = map.project(corner).y / zoom < 2;
      exceedsBottom = map.project(corner).y / Math.pow(2, zoom) >= 255;
    }
```

for some reason, in the map I use, the variable `map.project(corner).y` returns a negative value, and obviously this is always lower than 2

then next both checks return `True`:

`exceedsTop = map.project(corner).y < 2;`
or
`exceedsTop = map.project(corner).y / zoom < 2;`

and it make leave the normal execution of the handler function.

Im still not use why `map.project(corner).y` return a negative value, probably due the system coordinates Im using.

And probably this is a bug on the logics in the code. @jywarren

### LINKED ISSUE #1032: add exemption for L.CRS.Simple to _cornerExceedsMapLats()
### 🆕🐥☝ First Timers Only.

This issue is reserved for people who never contributed to Open Source before. We know that the process of creating a pull request is the biggest barrier for new contributors. This issue is for you 💝

[About First Timers Only](http://www.firsttimersonly.com/).

### 🤔 What you need to know.

Nothing. This issue is meant to welcome you to Open Source :) We are happy to walk you through the process.

### 📋 Step by Step

- [ ] 🙋 **Claim this issue**: Comment below.

  Once claimed we add you as contributor to this repository.

- [ ] 👌 **Accept our invitation** to this repository. Once accepted, assign yourself to this issue

- [ ] 📝 **Update** the file [\src/DistortableImageOverlay.js](https://github.com/publiclab/Leaflet.DistortableImage/blob/main/src%2FDistortableImageOverlay.js) in the `Leaflet.DistortableImage` repository (press the little pen Icon) and edit the line as shown below.

```diff
@@ -211,16 +211,18 @@ L.DistortableImageOverlay = L.ImageOverlay.extend({
   },
 
   _cornerExceedsMapLats: function(zoom, corner, map) {
+    if (map.options.crs == L.CRS.Simple) {
+      return false;
+    } else {
        var exceedsTop;
        var exceedsBottom;
        if (zoom === 0) {
          exceedsTop = map.project(corner).y < 2;
          exceedsBottom = map.project(corner).y >= 255;
        } else {
          exceedsTop = map.project(corner).y / zoom < 2;
          exceedsBottom = map.project(corner).y / Math.pow(2, zoom) >= 255;
      }
+    }
   },
 
   setCorners: function(latlngObj) {
```

- [ ] 💾 **Commit** your changes

- [ ] 🔀 **Start a Pull Request**. There are two ways how you can start a pull request:

1. If you are familiar with the terminal or would like to learn it, [here is a great tutorial](https://egghead.io/series/how-to-contribute-to-an-open-source-project-on-github) on how to send a pull request using the terminal.

2. You can [edit files directly in your browser](https://help.github.com/articles/editing-files-in-your-repository/)

- [ ] 🏁 **Done** Ask in comments for a review :)

### 🤔❓ Questions

Leave a comment below!

This issue was created by [First-Timers-Bot](https://github.com/hoodiehq/first-timers-bot).

### PR DESCRIPTION
**Pull request by Aman-Jamshed**
added exemption for L.CRS.Simple to _cornerExceedsMapLats()

Fixes #1032  (<=== Add issue number here)
Fixes #1007

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [ ] PR is descriptively titled 📑 and links the original issue above 🔗
* [ ] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `grunt test`
* [ ] code is in uniquely-named feature branch and has no merge conflicts 📁
* [ ] screenshots/GIFs are attached 📎 in case of UI updates
* [ ] @mention the original creator of the issue in a comment below for help or for a review

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software 

Thanks!

### CODE REVIEW

**Comment by Aman-Jamshed** [comment]
@jywarren Please check and let me know if something is wrong?

**Review by jywarren — APPROVED** [review]
This looks perfect. Thanks so much!!!

File: src/DistortableImageOverlay.js [hunk 1]
```diff
@@ -211,16 +211,20 @@ L.DistortableImageOverlay = L.ImageOverlay.extend({
...
-    var exceedsBottom;
-    if (zoom === 0) {
-      exceedsTop = map.project(corner).y < 2;
-      exceedsBottom = map.project(corner).y >= 255;
-    } else {
-      exceedsTop = map.project(corner).y / zoom < 2;
-      exceedsBottom = map.project(corner).y / Math.pow(2, zoom) >= 255;
+    if (map.options.crs == L.CRS.Simple) {
+      return false;
+    } else {
```
Comment (jywarren) [inline src/DistortableImageOverlay.js:216]: Aha, it's complaining in the automatic tests about trailing spaces. Try this quick fix!

```suggestion
    } else {
```

File: src/DistortableImageOverlay.js [hunk 2]
```diff
@@ -211,16 +211,20 @@ L.DistortableImageOverlay = L.ImageOverlay.extend({
...
+      var exceedsTop;
+      var exceedsBottom;
+      if (zoom === 0) {
+        exceedsTop = map.project(corner).y < 2;
+        exceedsBottom = map.project(corner).y >= 255;
+      } else {
+        exceedsTop = map.project(corner).y / zoom < 2;
+        exceedsBottom = map.project(corner).y / Math.pow(2, zoom) >= 255;
+      }
+      return (exceedsTop || exceedsBottom);
```
Comment (jywarren) [inline src/DistortableImageOverlay.js:226]:
```suggestion
      return (exceedsTop || exceedsBottom);
```

Here too. It's quite picky about formatting!

**Review by jywarren — CHANGES_REQUESTED** [review]
Can you try making these small adjustments, and one more thing - can you try running `grunt build` to compile the code, and then `git add dist/` to add the compiled files? That should complete this PR! ✨✨

**Comment by jywarren** [comment]
Great work!

**Review by jywarren — APPROVED** [review]

**Comment by jywarren** [comment]
Hi, i wanted to show how to add the dist files -- actually in GitPod, they're automatically built when you start up the gitpod environment -- using the link in the comment above, or https://gitpod.io#https://github.com/publiclab/Leaflet.DistortableImage/pull/1033.

You have to add it with `git add dist` OR click the little `+` button in the "Source Control" sidebar tab in Gitpod:

image

Then commit that. It contains the compiled version of the changes you'd made in the `/src/` directory. 

I'm going to go ahead and do it this time just to merge this! Thank you!

**Comment by jywarren** [comment]
Great work @Aman-Jamshed -- thank you!

**Comment by Aman-Jamshed** [comment]
Hey @jywarren Thankyou so much for guidance...Actually I was trying running the command ```grunt build``` as you said earlier to compile the code and I was getting grunt command not found. So I was trying to find its solution. Thanks for letting me know how its done in Gitpod. This is my first time using Gitpod. 

Could you please tell me what to do next?  I'm willing to contribute more and learn about this project.

**Comment by Aman-Jamshed** [comment]
@jywarren Please respond. It will be very kind of you!

**Comment by themacboy** [comment]
https://gruntjs.com/getting-started

Try this to install grunt as global:

npm install -g grunt-cli

**Comment by jywarren** [comment]
Hi @Aman-Jamshed i'm sorry I have been traveling so I missed this! The command is actually not `grunt build` but `npm run setup` -- sorry I messed that up. You can read more about how to set up the project here: https://github.com/publiclab/Leaflet.DistortableImage/blob/main/CONTRIBUTING.md#setup

Thank you very much @themacboy as well!

**Pull request merged**